src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,78 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+
 public class WorkWithFile {
+    private static final int NUMBER_ONE = 1;
+    private String processingResult = "";
+    private StringBuilder builderReadMethod;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readWithFile(fromFileName);
+        dataProcessing(builderReadMethod);
+        writeReportToFile(processingResult, toFileName);
+    }
+
+    private StringBuilder readWithFile(String fileName) {
+        builderReadMethod = new StringBuilder();
 
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName));
+            try {
+                String value = "";
+                while (value != null) {
+                    builderReadMethod.append(value).append(",");
+                    value = bufferedReader.readLine();
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't read this file", e);
+        }
+        return builderReadMethod;
+    }
+
+    private void dataProcessing(StringBuilder stringBuilder) {
+        String[] dataArray = stringBuilder.toString().split(",");
+        int supply = 0;
+        int buy = 0;
+        for (int i = 0; i < dataArray.length; i++) {
+            if (dataArray[i].equals("supply")) {
+                supply += Integer.parseInt(dataArray[i + NUMBER_ONE]);
```

Comment:
```suggestion
                supply += Integer.parseInt(dataArray[i + 1]);
```
constants are supposed to add a description, what a value means
NUMBER_ONE doesn't give any information on what is this value for

Reply:
В минулих завдання мені радил замінювати цифри в коді константантами . В цьому варіанті не варто чи питання тільки  в назві NUMBER_ONE?